morango/models/utils.py

```diff
@@ -0,0 +1,197 @@
+import hashlib
```

Comment:
This file is where a lot of the real "work" happens, to support calculation of the Instance ID.